### ISSUE TITLE
When has_one used, one item is returned but it's an array

### ISSUE THREAD

**Issue by jim-mackenzie** [rails-api/active_model_serializers#697]
Currently

class UserSerializer < ActiveModelSerializer
  attributes :username
  has_one :pet
end

currently returns

{ username: ‘bsmith’, pet: [{name: ‘fluffy’}]}

but this is expected

{ username: ‘bsmith’, pet: {name: ‘fluffy’}}

**Comment by r4m** [comment]
I've investigated a little bit and it seems that this bug comes up when setting `config.eager_load = true` in a Rails env config.

**Comment by joaomdmoura** [comment]
This is weird, please @r4m share any updates you get in here.
0.8.x and 0.9.x support is iced for now because we are focused on 0.10.x but we might be able to check it again in a while.

**Comment by r4m** [comment]
@joaomdmoura I spent a couple of hours on it. I discovered that it happens when `config.eager_load = true` and the `embed :objects, embed_in_root: false` call is used inside a specific serializer. In this situation, the `embed(type, options={})` method inside the `active_model_serializers/lib/active_model/serializer.rb` is called multiple times, completely overriding the global configuration of AMS, let say an AMS installation defined as:

```ruby
ActiveModel::Serializer.setup do |config|
  config.embed = :ids
  config.embed_in_root = true
end
````

So a workaround is removing any custom "embed" call in the serializers. However this not solve the error. Looking at the code of AMS I've seen that there is a local var `@wrap_in_array` that determines the way in which an association is serialized. The problem is that the wrap in array var affects not only those data that need to be embedded in root, but also those that are nested serialization. That causes the error of having nested data that are supposed to be single hash instances or nil values and instead the becomes array of hashes or empty array.
You can check it looking into the `as_json(options={})` method of `active_model_serializers/lib/active_model/default_serializer.rb` and into the `serializable_object(options={}) method of active_model_serializers/lib/active_model/serializer.rb`.

**Comment by padi** [comment]
@r4m the bug is happening to me even when I don't ever use `embed` inside any specific serializer. Will still have to triangulate what specifically causes the error though.

@remear what's the reason for closing this? I want to know if it's still worth the effort to resolve this bug (cloned 0.9.0 recently).

**Comment by remear** [comment]
Age, and lack of activity. We're focussing on finalizing 0.10 and I don't expect this to get much attention. If someone has interest in submitting a PR for this they're welcome to do so. Though, I'd suggest everyone migrate to 0.10 and help us fix the remaining 0.10 issues.

**Comment by localredhead** [comment]
+1
